### DISCUSSION TITLE
Could not resolve "stream" and "crypto" in quasar-cli with vite

### DISCUSSION THREAD

**Discussion by icebergtsn** [quasarframework/quasar#14796] [Quasar App Extensions]
I used quasar webpack to write extensions manifest v2 before, but quasar webpack does not support manifest v3, so I switched to vite, unable to get steam and crypto, webpack had a solution before, but I didn't find an effective solution in vite

webpack:[https://quasar.dev/start/upgrade-guide#nodejs-polyfills](https://quasar.dev/start/upgrade-guide#nodejs-polyfills)
vite:?

error
============================

✘ [ERROR] Could not resolve "stream"

    node_modules/@metamask/obs-store/dist/asStream.js:4:25:
      4 │ const stream_1 = require("stream");
        ╵                          ~~~~~~~~

  The package "stream" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "platform: 'node'" to do that, which will remove this error.

✘ [ERROR] Could not resolve "crypto"

    node_modules/ethereumjs-wallet/dist.browser/thirdparty.js:62:34:
      62 │ var crypto = __importStar(require("crypto"));
         ╵                                   ~~~~~~~~

  The package "crypto" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "platform: 'node'" to do that, which will remove this error.

node_modules/@metamask/obs-store/dist/asStream.js:4:25: ERROR: Could not resolve "stream"
node_modules/cipher-base/index.js:2:24: ERROR: Could not resolve "stream"
node_modules/ethereumjs-wallet/dist.browser/index.js:73:34: ERROR: Could not resolve "crypto"
node_modules/ethereumjs-wallet/dist.browser/thirdparty.js:62:34: ERROR: Could not resolve "crypto"

====================================

**Comment by TobyMosque** [comment]
just extend the viteConfig to include the needed polyfills
https://quasar.dev/quasar-cli-vite/handling-vite#usage-with-quasar-config-js
https://gist.github.com/FbN/0e651105937c8000f10fefdf9ec9af3d

**Reply by icebergtsn** [reply]
First of all, thank you for such a quick and precise explanation.

But I encountered a problem, how to convert vite.config to quasar.config, I tried to put vite content directly in extendViteConf, I also tried to take out plugins and put it in vitePlugins, but it didn't work, or it was the way I used something wrong.

Could you take a moment and show how to use this configuration in quasar.config.ts? That would be greatly appreciated.

**Reply by qq815776412** [reply]
> just extend the viteConfig to include the needed polyfills https://quasar.dev/quasar-cli-vite/handling-vite#usage-with-quasar-config-js https://gist.github.com/FbN/0e651105937c8000f10fefdf9ec9af3d

<img width="703" alt="截屏2022-11-16 13 51 48" src="https://user-images.githubusercontent.com/8295680/202095138-9aef0d46-666d-4063-860b-c111df317143.png">
<img width="738" alt="截屏2022-11-16 13 52 34" src="https://user-images.githubusercontent.com/8295680/202095144-852842a4-2e86-4e44-aec6-c73e6ba7e24d.png">
